File: mingachevir-real-estate-agency-server/src/main/java/io/mingachevir/mingachevirrealestateserver/model/dto/admin/ActiveCategoriesDTO.java
package io.mingachevir.mingachevirrealestateserver.model.dto.admin;

import lombok.Data;

import java.util.List;

@Data
public class ActiveCategoriesDTO {
    private List<MainCategoryDTO> mainCategories;
    private List<SubCategoryDTO> subCategories;
    private List<ParametersDTO> parameters;
    private List<SelectiveParameterValueDTO> selectiveParameters;

    public ActiveCategoriesDTO(List<MainCategoryDTO> mainCategories, List<SubCategoryDTO> subCategories,
                               List<ParametersDTO> parameters, List<SelectiveParameterValueDTO> selectiveParameters) {
        this.mainCategories = mainCategories;
        this.subCategories = subCategories;
        this.parameters = parameters;
        this.selectiveParameters = selectiveParameters;
    }
}

File: ngx-admin/src/app/model/category/ActiveCategoriesDTO.ts
import {MainCategoryDTO} from "./MainCategoryDTO";
import {SubCategoryDTO} from "./SubCategoryDTO";
import {ParametersDTO} from './ParametersDTO';
import {SelectiveParameterValueDTO} from './SelectiveParameterValueDTO';

export interface ActiveCategoriesDTO {
  mainCategories: MainCategoryDTO[];
  subCategories: SubCategoryDTO[];
  parameters: ParametersDTO[];
  selectiveParameters: SelectiveParameterValueDTO[];
}
